### ISSUE TITLE
LT02 forces a line break before `LATERAL` keyword of the first lateral join only when a second sibling join follows it

### ISSUE THREAD

**Issue by Prometheus3375** [sqlfluff/sqlfluff#8345]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When a `FROM` clause's bracketed subquery is followed by 2+ consecutive `CROSS JOIN LATERAL` clauses, `LT02` requires the *first* `LATERAL` to be pushed onto its own indented line, but has no complaint about the *second* one staying on the same line as `CROSS JOIN`. If only one `JOIN LATERAL` clause is present, it's allowed to stay on one line with no violation at all.

The content of the two join clauses doesn't matter -- swapping which one comes first just moves the forced break to whichever one is now first.

### Expected Behaviour

`CROSS JOIN LATERAL` stays on the same line regardless of how many such statements are present.

### Observed Behavior

Keyword `LATERAL` of the first join is pushed to the next line with subsequent clause contents indented.

### How to reproduce

Lint is passed with single join:
```sql
SELECT *
FROM (
    SELECT 1 AS id
) t
    CROSS JOIN LATERAL (
        SELECT 1 AS x
    ) a
```

```
$ sqlfluff lint --dialect postgres min_single.sql --rules LT02
All Finished!
```

List isn't passed with multiple joins:
```sql
SELECT *
FROM (
    SELECT 1 AS id
) t
    CROSS JOIN LATERAL (
        SELECT 1 AS x
    ) a
    CROSS JOIN LATERAL (
        SELECT 1 AS y
    ) b
```

```
$ sqlfluff lint --dialect postgres min.sql --rules LT02
L:   5 | P:  15 | LT02 | Expected line break and indent of 8 spaces before
                       | 'LATERAL'. [layout.indent]
L:   6 | P:   1 | LT02 | Expected indent of 12 spaces. [layout.indent]
L:   7 | P:   1 | LT02 | Expected indent of 8 spaces. [layout.indent]
```

### Dialect

`postgres`

### Version

sqlfluff: 4.2.2
Python: 3.12.10

### Configuration

```ini
[sqlfluff]
dialect = postgres

[sqlfluff:indentation]
indented_joins = True
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by aryansk** [comment]
I've opened a draft PR with a fix: https://github.com/sqlfluff/sqlfluff/pull/8348 — root cause is in the reflow reindent logic (untaken join-body indent exemption only covered brackets immediately following the indent, so `JOIN LATERAL (` was missed).

**Comment by apoorva-01** [comment]
Traced the forced break to the bracket special-case in `_map_line_buffers` (`src/sqlfluff/utils/reflow/reindent.py`): it only skips the untaken indent when the next element is a `start_bracket`. `WHERE (` is fine, `CROSS JOIN LATERAL (` isn't, because `LATERAL` sits between the indent and the bracket.

Looking past the keyword to the bracket fixes the first join but not the second, and I couldn't get them consistent. The real blocker: I don't know the target. Your single-join example doesn't lint clean on `main` for me either (`L:5 P:1 | Line should not be indented`), so this looks tied to `indented_joins`, not just the `LATERAL` break.

@peterbud what should two-plus `CROSS JOIN LATERAL` blocks look like with `indented_joins` on and off? Once I know, I'll open a PR.

**Comment by peterbud** [comment]
Hi @apoorva-01 : the first example does pass the linter, if you configure `indented_joins` as @Prometheus3375 described.

I _guess_ the target should be as @Prometheus3375  described, basically treating `LITERAL` as part of the `CROSS JOIN` construct.

```sql
SELECT *
FROM (
    SELECT 1 AS id
) t
    CROSS JOIN LATERAL (
        SELECT 1 AS x
    ) a
    CROSS JOIN LATERAL (
        SELECT 1 AS y
    ) b
```

Somehow also related that _without_ `LATERAL` the simple CROSS JOIN cosntruct would trigger `ST05` and restructure these subqueries to CTE:
`select_statement clauses should not contain subqueries.
                       | Use CTEs instead [structure.subquery]
`

So I'm not sure how much sense it would make to spend efforts on trying to fix the indentation TBH.
